pkg/ctl/create/nodegroup.go

```diff
@@ -70,71 +104,205 @@ func doCreateNodeGroup(p *api.ProviderConfig, cfg *api.ClusterConfig, ng *api.No
 		return err
 	}
 
-	if !ctl.IsSupportedRegion() {
-		return cmdutils.ErrUnsupportedRegion(p)
-	}
-	logger.Info("using region %s", meta.Region)
+	var stackManager *manager.StackCollection
 
-	if err := ctl.CheckAuth(); err != nil {
-		return err
-	}
+	if clusterConfigFile != "" {
+		if err := eks.LoadConfigFromFile(clusterConfigFile, cfg); err != nil {
+			return err
+		}
+		meta = cfg.Metadata
+
+		// Limit nodegroups to set specified on command line via globs
+		filterNodeGroups(cfg)
+
+		incompatibleFlags := []string{
```

Comment:
Any chance the set of validations starting here could be moved into it's own function?

Reply:
Sure. In this case, though, I was again following the pattern in cluster.go... For the most part, I tried to maintain a consistent style. I'm sympathetic to pulling some of these things out, perhaps across the board in a refactoring pass?

Reply:
Issue [#399](https://github.com/weaveworks/eksctl/issues/399) was already opened on Jan 9th for this issue.